### PR TITLE
Add check for uncompressed data

### PR DESCRIPTION
**Pull request by marxin**
The PR verifies that the uncompressed data matches the declared size in the compression section header.

### CODE REVIEW

File: src/read/mod.rs [hunk 1]
```diff
@@ -1003,13 +979,36 @@ impl<'data> CompressedData<'data> {
...
+                        decoder
+                            .read_to_end(&mut decompressed)
+                            .ok()
+                            .read_error("Invalid zstd compressed data")?;
+                    }
+                    _ => unreachable!(),
+                }
+                if size != decompressed.len() {
+                    return Err(Error(
+                        "Uncompressed size does not match the uncompressed size declared in the compression header",
```
Comment (philipc) [inline src/read/mod.rs:1008]:
```suggestion
                        "Uncompressed data size does not match compression header",
```

**Review by philipc — COMMENTED** [review]
Did you encounter a file this occurs for?

**Comment by marxin** [comment]
No, but I was surprised it was not reported when I wrote a unit test for `CompressedData::decompress`.

**Review by philipc — APPROVED** [review]
Thanks!

**Pull request merged**